components/resolved_ts/src/scanner.rs

```diff
@@ -0,0 +1,227 @@
+                        .range(None, None)
+                        .build_delta_scanner(task.checkpoint_ts, txn_extra_op)
+                        .unwrap();
+                    let mut done = false;
+                    while !done && !(task.cancelled)() {
+                        let (es, has_remaining) = match Self::scan_delta(&mut scanner) {
+                            Ok(rs) => rs,
+                            Err(e) => {
+                                warn!("resolved_ts scan delta failed"; "err" => ?e);
+                                let ScanTask {
+                                    on_error,
+                                    region,
+                                    id,
+                                    ..
+                                } = task;
+                                if let Some(on_error) = on_error {
+                                    on_error(id, region, e);
+                                }
+                                return;
+                            }
+                        };
+                        done = !has_remaining;
+                        entries.push(ScanEntry::TxnEntry(es));
+                    }
+                }
+                ScanMode::LockOnly => {
+                    let mut reader = MvccReader::new(
+                        snap,
+                        Some(MvccScanMode::Forward),
+                        false,
+                        IsolationLevel::Si,
+                    );
+                    let mut done = false;
+                    let mut start = None;
+                    while !done && !(task.cancelled)() {
+                        let (locks, has_remaining) =
+                            match Self::scan_locks(&mut reader, start.as_ref(), task.checkpoint_ts)
+                            {
+                                Ok(rs) => rs,
+                                Err(e) => {
+                                    warn!("resolved_ts scan delta failed"; "err" => ?e);
+                                    let ScanTask {
+                                        on_error,
+                                        region,
+                                        id,
+                                        ..
+                                    } = task;
+                                    if let Some(on_error) = on_error {
+                                        on_error(id, region, e);
+                                    }
+                                    return;
+                                }
+                            };
+                        done = !has_remaining;
+                        if has_remaining {
+                            start = Some(locks.last().unwrap().0.clone())
+                        }
+                        entries.push(ScanEntry::Lock(locks));
+                    }
+                }
+            }
+            entries.push(ScanEntry::None);
```

Comment:
Is `ScanEntry::None` serve as an ending flag? It may unnecessary because we can know when a `vec` end.

Reply:
It was needed for cdc scan